spec/requests/active_record_request_spec.rb

```diff
@@ -0,0 +1,163 @@
+              { name: "ActiveBars", kind: "EntitySet", url: "ActiveBars" },
+              { name: "Fake data" }
+            ]
+          }.to_json)
+        end
+      end
+    end
+
+    context "$metadata" do
+      let(:path) { "#{root}/$metadata" }
+
+      it "renders as expected" do
+        expect_output(file_fixture("metadata_basic.xml").read)
+      end
+
+      context "with hook" do
+        let(:options) do
+          {
+            transform_schema_for_metadata: lambda do |schema|
+              schema.namespace = "Test2"
+              schema.entity_types["ActiveFoo"].properties["Name"] = SimpleProperty.new("Name")
+              schema
+            end
+          }
+        end
+
+        it "renders as expected" do
+          expect_output(file_fixture("metadata_after_hook.xml").read)
+        end
+      end
+    end
+
+    context "resource" do
+      let(:path) { "#{root}/ActiveFoos" }
+
+      before do
+        (1..2).each do |n|
+          create(:active_foo, name: "test #{n}")
+        end
+      end
+
+      it "renders as expected" do
+        expect_output({
+          "@odata.context": "http://www.example.com/odata/$metadata#ActiveFoos",
+          value: [
+            { Id: 1, Name: "test 1", CreatedAt: "2020-01-01T12:00:00Z", UpdatedAt: "2020-01-01T12:00:00Z" },
+            { Id: 2, Name: "test 2", CreatedAt: "2020-01-01T12:00:00Z", UpdatedAt: "2020-01-01T12:00:00Z" }
+          ]
+        }.to_json)
+      end
+
+      context "with feed hook" do
+        let(:options) do
+          {
+            transform_json_for_resource_feed: lambda do |json|
+              json[:value].push({ name: "Fake data" })
+              json
+            end
+          }
+        end
+
+        it "renders as expected" do
+          expect_output({
+            "@odata.context": "http://www.example.com/odata/$metadata#ActiveFoos",
+            value: [
+              { Id: 1, Name: "test 1", CreatedAt: "2020-01-01T12:00:00Z", UpdatedAt: "2020-01-01T12:00:00Z" },
+              { Id: 2, Name: "test 2", CreatedAt: "2020-01-01T12:00:00Z", UpdatedAt: "2020-01-01T12:00:00Z" },
+              { name: "Fake data" }
+            ]
+          }.to_json)
+        end
+      end
+
+      context "with entry hook" do
+        let(:options) do
+          {
+            transform_json_for_resource_entry: lambda do |json|
+              json["Name"] = "foo"
+              json
+            end
+          }
+        end
+
+        it "renders as expected" do
+          expect_output({
+            "@odata.context": "http://www.example.com/odata/$metadata#ActiveFoos",
+            value: [
+              { Id: 1, Name: "foo", CreatedAt: "2020-01-01T12:00:00Z", UpdatedAt: "2020-01-01T12:00:00Z" },
+              { Id: 2, Name: "foo", CreatedAt: "2020-01-01T12:00:00Z", UpdatedAt: "2020-01-01T12:00:00Z" },
+            ]
+          }.to_json)
+        end
+      end
+    end
+  end
+end
```

Comment:
This is a really nice spec. I read it and it immediately reassured me that this thing is doing what it's supposed to. Great work.